pandas/tests/plotting/test_hist_method.py

```diff
@@ -293,6 +315,33 @@ def test_hist_column_order_unchanged(self, column, expected):
 
         assert result == expected
 
+    @pytest.mark.parametrize("by", [None, "c"])
+    @pytest.mark.parametrize("column", [None, "b"])
+    @pytest.mark.parametrize("label", [None, "d"])
+    def test_hist_with_legend(self, by, column, label):
+        # GH 6279
+        # Histogram can have a legend
+        expected_axes_num = 1 if by is None and column is not None else 2
+        expected_layout = (1, expected_axes_num)
+        expected_labels = label or column or ["a", "b"]
+        if by is not None:
+            expected_labels = [expected_labels] * 2
+
+        index = Index(15 * ["1"] + 15 * ["2"], name="c")
+        df = DataFrame(np.random.randn(30, 2), index=index, columns=["a", "b"])
+
+        kwargs = {"legend": True, "by": by, "column": column}
+        # We get warnings if kwargs contains "label": None
+        if label is not None:
+            kwargs["label"] = label
+
+        axes = _check_plot_works(df.hist, **kwargs)
+        self._check_axes_shape(axes, axes_num=expected_axes_num, layout=expected_layout)
+        if by is None and column is None and label is None:
+            axes = axes[0]
```

Comment:
why do we need to do this?

Reply:
A good question - something I wondered myself when parameterizing the tests but considered out of scope. The return value from calling ``df.hist(**kwargs)`` here is:

    array([[<matplotlib.axes._subplots.AxesSubplot object at 0x7fa1dc0a6950>,
        <matplotlib.axes._subplots.AxesSubplot object at 0x7fa1dc0c0210>]],
      dtype=object)

By extracting the first element, puts us on similar footing as the other tests, e.g. when by='c':

    array([<matplotlib.axes._subplots.AxesSubplot object at 0x7fa1d754fe90>,
       <matplotlib.axes._subplots.AxesSubplot object at 0x7fa1d7564b90>],
      dtype=object)

but I do not know why the first call produces the [[ - ]] result.

Reply:
yeah, just recall this will happen if simply do `df.hist()` i think. nvm, it's fine this way.